### DISCUSSION TITLE
Output does not consistently include image links and urls

### DISCUSSION THREAD

**Discussion by M-NRVA** [dsdanielpark/Bard-API#209] [General]
The output using the API sometimes includes links for both the images (usually displayed in the Web UI) and URLs, but some other times only says [Link to **] or [Image of**]. Does anyone have a sense for why this inconsistency and if there is any way to make bard more reliably provide links in the output? Thanks!

**Comment by dsdanielpark** [comment]
Hello, thank you for your support of the package.

The functions for link and image extraction are wrapped in the utils.py file. The code is designed to extract this content from the returned structure, but if the response from Bard is empty, it may not be matched and returned. If you provide specific prompts or examples, we will be happy to investigate. Alternatively, if you could contribute by modifying the following function and creating a Pull Request, we would greatly appreciate it.
[Link to the code](https://github.com/dsdanielpark/Bard-API/blob/main/bardapi/utils.py#L9).

**Reply by M-NRVA** [reply]
Thanks for the quick response and I will take a look. Including an extreme example (because the prompt asks for product links). Same query through the API vs Web, run within seconds of each other.

![Screenshot 2023-10-14 at 7 34 02 PM](https://github.com/dsdanielpark/Bard-API/assets/145135058/523b5532-7d22-49cf-87a8-86a20b05369f)

![Screenshot 2023-10-14 at 7 34 12 PM](https://github.com/dsdanielpark/Bard-API/assets/145135058/0e37fbe5-d58e-48d9-b3c3-6f86b7aa801d)
